src/panels/config/devices/ha-devices-data-table.ts

```diff
@@ -145,7 +145,7 @@ export class HaDevicesDataTable extends LitElement {
                 return html`
                   ${name}<br />
                   ${device.area} | ${device.integration}<br />
-                  ${battery
+                  ${battery && !isNaN(battery.state as any)
```

Comment:
`battery.state` is always a string. You will need to parse it first. But I prefer that you compare it against `unavailable` and `unknown` instead

Reply:
Or this:

```suggestion
                  ${battery && !isNaN(Number(battery.state))
```

Reply:
`isNaN` does a string to number conversion, but I can also check for `unavailable` and `unknown`

Reply:
in that case it's fine.